backend_new/src/services/listing.service.ts

```diff
@@ -0,0 +1,336 @@
+    reservedCommunityTypes: true,
+    listingImages: {
+      include: {
+        assets: true,
+      },
+    },
+    listingMultiselectQuestions: {
+      include: {
+        multiselectQuestions: true,
+      },
+    },
+    listingFeatures: true,
+    listingUtilities: true,
+  },
+};
+
+views.base = {
+  ...views.fundamentals,
+  units: {
+    include: {
+      unitTypes: true,
+      unitAmiChartOverrides: true,
+      amiChart: {
+        include: {
+          amiChartItem: true,
+        },
+      },
+    },
+  },
+};
+
+views.full = {
+  ...views.fundamentals,
+  applicationMethods: {
+    include: {
+      paperApplications: {
+        include: {
+          assets: true,
+        },
+      },
+    },
+  },
+  listingsBuildingSelectionCriteriaFile: true,
+  listingEvents: {
+    include: {
+      assets: true,
+    },
+  },
+  listingsResult: true,
+  listingsLeasingAgentAddress: true,
+  listingsApplicationPickUpAddress: true,
+  listingsApplicationDropOffAddress: true,
+  units: {
+    include: {
+      unitAmiChartOverrides: true,
+      unitTypes: true,
+      unitRentTypes: true,
+      unitAccessibilityPriorityTypes: true,
+      amiChart: {
+        include: {
+          jurisdictions: true,
+          amiChartItem: true,
+          unitGroupAmiLevels: true,
+        },
+      },
+    },
+  },
+};
+
+views.details = {
+  ...views.base,
+  ...views.full,
+};
+
+/*
+  this is the service for listings
+  it handles all the backend's business logic for reading in listing(s)
+*/
+@Injectable()
+export class ListingService {
+  constructor(private prisma: PrismaService) {}
+
+  /*
+    this will get a set of listings given the params passed in
+    this set can either be paginated or not depending on the params
+    it will return both the set of listings, and some meta information to help with pagination
+  */
+  async list(params: ListingsQueryParams) {
+    const whereClause = this.buildWhereClause(params.filter, params.search);
+    const isPaginated = shouldPaginate(params.limit, params.page);
+
+    const count = await this.prisma.listings.count({
+      where: whereClause,
+    });
+
+    const listingsRaw = await this.prisma.listings.findMany({
+      skip: calculateSkip(params.limit, params.page),
+      take: calculateTake(params.limit),
+      orderBy: buildOrderBy(params.orderBy, params.orderDir),
+      include: views[params.view ?? 'full'],
+      where: whereClause,
+    });
+
+    const listings = mapTo(ListingGet, listingsRaw);
+
+    listings.forEach((listing) => {
+      if (Array.isArray(listing.units) && listing.units.length > 0) {
+        listing.unitsSummarized = {
+          byUnitTypeAndRent: summarizeUnitsByTypeAndRent(
+            listing.units,
+            listing,
+          ),
+        };
+      }
+    });
+
+    const itemsPerPage =
+      isPaginated && params.limit !== 'all' ? params.limit : listings.length;
+    const totalItems = isPaginated ? count : listings.length;
```

Comment:
Is this ternary operator necessary? Wouldn't `count` always be the totalItems even if it's not paginated? Doesn't look like you are doing any pagination specific things in the count query